siliconcompiler/scheduler/__init__.py

```diff
@@ -121,6 +124,8 @@ def run(chip):
     # Merge cfgs from last executed tasks, and write out a final manifest.
     _finalize_run(chip)
 
+    _cache_clear()
+
```

Comment:
If you move this into _finalize_run, I think that would be sufficient to ensure some of the other callers of `run` will hit this.